jwst/photom/photom.py

```diff
@@ -425,6 +425,9 @@ def calc_niriss(self, ftab):
             raise DataModelTypeError(f"Unexpected input data model type for NIRISS: {self.input.__class__.__name__}")
 
         elif self.exptype in ['NIS_SOSS']:
+            if isinstance(self.input, datamodels.ImageModel):
+                raise DataModelTypeError(f"Unexpected input data model type for NIRISS: {self.input.__class__.__name__}")
+
```

Comment:
So remind me what happens when this error is raised. Does it halt the whole pipeline flow, or does it just emit an error message and then return control to the `Spec2` pipeline? We don't like to have things running in operations just abort mid-stream.

Reply:
And while it's worthwhile to have this kind of check in the `photom` step itself, incase someone were to try running `photom` by itself on an invalid input, would it also make sense to have a hook up in the `Spec2` pipeline that just skips the `photom` step when it knows `extract_1d` was also skipped?

Reply:
sure, I can do that.  It seems like that should probably have already been done in the PR where this `DataModelTypeError` was first implemented, but better late than never I suppose